react/src/components/DesktopAppDownloadModal.tsx

```diff
@@ -0,0 +1,91 @@
+import { useSuspendedBackendaiClient } from '../hooks';
+import BAIModal, { BAIModalProps } from './BAIModal';
+import Flex from './Flex';
+import { Select, Button, Descriptions, Tooltip } from 'antd';
+import _ from 'lodash';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+const APP_DOWNLOAD_MAP = {
+  Linux: {
+    os: 'linux',
+    architecture: ['arm64', 'x64'],
+    extension: 'zip',
+  },
+  MacOS: {
+    os: 'macos',
+    architecture: ['arm64', 'x64'],
+    extension: 'dmg',
+  },
+  Windows: {
+    os: 'win32',
+    architecture: ['arm64', 'x64'],
+    extension: 'zip',
+  },
+};
+
+interface DesktopAppDownloadModalProps extends BAIModalProps {
+  onRequestClose?: () => void;
+}
+
+const DesktopAppDownloadModal: React.FC<DesktopAppDownloadModalProps> = ({
+  onRequestClose,
+  ...baiModalProps
+}) => {
+  const { t } = useTranslation();
+  const baiClient = useSuspendedBackendaiClient();
+  const appDownloadUrl = baiClient?._config?.appDownloadUrl || '';
+  const [selectedOS, setSelectedOS] =
+    useState<keyof typeof APP_DOWNLOAD_MAP>('MacOS');
+
+  const handleDownload = (architecture: string) => {
+    // @ts-ignore
+    const pkgVersion = globalThis.packageVersion;
```

Comment:
I recommend that create and use a hook for APP_DOWNLOAD_MAP to hide most of details related APP_DOWNLOAD_MAP. For example, selectedOS and globalThis accessor, and string manipulation codes below (os, extension, downloadLink). We can separate and hide it from the components

Reply:
Additionally, it would be better to handle `architecture` in the same way.
```
const downloadLink = useAppDownloadLink(targetOS, architecture)
```